examples/python/kMeanClusteringSciPy.py

```diff
@@ -0,0 +1,132 @@
+			startTime = event.getTimestamp().toNanos()
+			syscallName = eventName[len(layout.eventSyscallEntryPrefix()):]
+
+			syscallInfo = [startTime, syscallName]
+			mapInitialInfo.put(tid, syscallInfo)
+		elif (eventName.startswith(layout.eventSyscallExitPrefix())):
+			tid = org.eclipse.tracecompass.analysis.os.linux.core.kernel.KernelTidAspect.INSTANCE.resolve(event)
+			endTime = event.getTimestamp().toNanos()
+			syscallInfo = mapInitialInfo.remove(tid)
+			if not(syscallInfo is None):
+				durations.append(float(endTime - syscallInfo[0]))
+				syscallNameDistinct, syscallNameCount, index = distinctSyscallCount(syscallInfo[1], syscallNameDistinct, syscallNameCount, syscallNameIndex)
+				syscallNameIndex = index
+				syscallNameList.append(syscallInfo[1])
+	return syscallNameDistinct, syscallNameCount, syscallNameList, durations
+
+# Gets the correct cluster the point belongs to
+def nearestPointIndex(value, codebook):
+	codebookNumpy = numpy.asarray(codebook)
+	lengths = numpy.abs(codebook - value)
+	return lengths.argmin()
+
+# Prints clustering statistics (percentage)
+def printStatistics():
+	iter = 0
+	for center in nditer(codebook):
+		print("For center of " + str(center))
+		percentage = float(len(durationData[iter]))/float(len(durations))*100
+		print("Percentage of syscalls with this cluster: "+ str(percentage))
+		iter += 1
+	for k, v in syscallNameDistinct.items():
+		print("Syscall name: " + k)
+		print("Count: " + str(v))
+
+# Run clustering algorithm on aspects
+def runClustering(durations):
+	codebook, distortion = kmeans(durations, int(argv[0]))
+	return codebook, distortion
+
+# Colors
+colors = numpy.array(["b","r","g","y","c","m","k"])
+
+# Prepare XY scatter points with color for matplotlib
+def prepareScatterPlot(syscallNameCount, codebook, syscallNameList):
+
+	durationData = [[] for x in range(int(argv[0])) ]
+	syscallNameIndex = [[] for x in range(len(durationData))]
+
+	for x in range(len(durations)):
+		index = nearestPointIndex(durations[x-1], codebook)
+		durationData[index].append(durations[x-1])
+		syscallNameIndex[index].append(syscallNameCount.get(str(syscallNameList[x-1])))
+
+	syscallName = ["" for x in range(len(syscallNameCount))]
+
+	for k, v in syscallNameCount.items():
+		syscallName[v] = k
+
+	return syscallNameIndex, syscallName, durationData
+
+# Format durations with commas (for thousands, millions)
+def formatDuration(x, pos):
+	return format(x, ",.0f")
+
+# Shows the xy plot with a matplotlib scatter plot
+def showPlot(syscallNameDistinct, syscallNameCount, durations, syscallNameList):
+	fig, ax = pyplot.subplots()
+	formatter = FuncFormatter(formatDuration)
+	ax.xaxis.set_major_formatter(formatter)
+	# Titles
+	pyplot.title("System Call Clustering by Duration", fontsize=20,fontweight='bold')
+	pyplot.xlabel("System Call Duration (ns)", fontsize=14)
+	pyplot.ylabel("System Call Name", fontsize=14)
+
+	for i in range(int(argv[0])):
```

Comment:
A good practice would be to start the script by verifying that an argument is present and print an error message if there is none, explaining the argument required.